functors/identity-functor.amb

```diff
@@ -0,0 +1,51 @@
+string_length[
+  in_ call.open call.(
+    func[
+      str[in_ arg.open arg.in int.in length.open_]|
+      int[
+        length[in_ str.open str]|
+        in_ str
+      ]|
+      open_
+    ]|
+    open return.open_
+  )
+]|
+map_identity[
+  in_ call.open call.(
+    func[
+      id[in_ arg.open arg.in arg.open_]|
+      arg[in_ id.open id.in func.in arg.open_]|
+      func[in_ func.in_ arg.open func.in identity.open_]|
+      identity[
+        in_ func.open func.open func
+      ]|
+      open_
+    ]|
+    open return.open_
+  )
+]|
+program[
+  out_ call.in_ map_identity|
+  out_ call.in_ string_length|
+  open func.open_|
+  
+  call[
+    out program.in map_identity.open_.return[open_.in program.in func]
+  ]|
+  call[
+    out program.in string_length.open_.return[open_.in program.in func.in func]
+  ]|
+  func[in_ map_identity.open map_identity.(
+    arg[
+      string[hello[]]|
+      in id.open_]|
```

Comment:
Could we wrap the arg to an `identity` and `open` it once in `id` to address the earlier feedback on lacking the explicit `identity` here?